packages/hop-node/src/aws/AwsSigner.ts

```diff
@@ -0,0 +1,123 @@
+import { BigNumber, Signer, providers } from 'ethers'
+import {
+  defineReadOnly,
+  getAddress as checksumAddress,
+  hashMessage,
+  parseTransaction,
+  keccak256,
+  recoverAddress,
+  resolveProperties,
+  serializeTransaction,
+  splitSignature
+} from 'ethers/lib/utils'
+import * as asn1 from 'asn1.js'
+
+const EcdsaPubKey = asn1.define('EcdsaPubKey', function () {
+  this.seq().obj(
+    this.key('algo').seq().obj(
+      this.key('a').objid(),
+      this.key('b').objid()
+    ),
+    this.key('pubKey').bitstr()
+  )
+})
+
+const EcdsaSigAsnParse = asn1.define('EcdsaSig', function () {
+  this.seq().obj(
+    this.key('r').int(),
+    this.key('s').int()
+  )
+})
+
+export type AwsSignerConfig = {
+  keyId: string
+  region?: string
+}
+
+// details:
+// https://ethereum.stackexchange.com/a/73371/5093
+// https://luhenning.medium.com/the-dark-side-of-the-elliptic-curve-signing-ethereum-transactions-with-aws-kms-in-javascript-83610d9a6f81
+// https://github.com/lucashenning/aws-kms-ethereum-signing/blob/master/aws-kms-sign.ts
+export abstract class AwsSigner extends Signer {
```

Comment:
I just thought about this one a bit. My current thinking is to leave it as is for two reasons:

* Technically, it extends a `Signer` class, so by definition, it is a `Signer` (if I'm thinking about that correctly)
* It is marked `abstract`, so can't be used standalone

Given that, I think it makes sense to leave it. If it was a `util` class, then I think it would be odd for the parent signing classes to extend `xxxUtil` instead of a signer. But very loose opinion here, so LMK what you think.